Libraries/LibJS/Runtime/Object.cpp

```diff
@@ -40,6 +40,46 @@
 
 namespace JS {
 
+PropertyDescriptor PropertyDescriptor::from_object(Interpreter& interpreter, const Object& object)
```

Comment:
Perhaps we could call this `from_dictionary`? `from_object` sounds like it's doing something slightly different :)